### ISSUE TITLE
QRO drive power

### ISSUE THREAD

**Issue by MolKa51** [drexjj/sbitx#72]
Is there a way to set the QRO drive level per band when QRO is enabled/used because the output in wattage is different on each band and can overdrive the external amplifier.
Gyula HA3HZ

**Comment by drexjj** [comment]
Hi Could you please clarify your request? Are you asking that each band does not exceed a certain power level such as 10 watts output power when using QRO?

**Comment by MolKa51** [comment]
Re: [drexjj/sbitx] QRO drive power (Issue #72)



Hello jj,

Yes, exactly that. I don't want to overdrive the output stage, because it's more difficult to limit the current there. I use the basic device with current limitation, so I haven't had to change the final stage FET so far. Drive=50 at 7MHz, drive=85 at 14MHz and drive=95 at 21MHz can be the maximum, then I will not exceed the maximum current consumption. For the 10W, I didn't look at where the drive should be set, I think that a different drive state is appropriate for everyone.

Sunday, September 29, 2024, 6:40:02 AM, you wrote:





Hi Could you please clarify your request? Are you asking that each band does not exceed a certain power level such as 10 watts output power when using QRO?
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
Best regards,
Molnar Gyula

**Comment by drexjj** [comment]
How many watts are you seeing at these levels?

Drive=50 at 7MHz
Drive=85 at 14MH
Drive=95 at 21MHz

**Comment by MolKa51** [comment]
Re: [drexjj/sbitx] QRO drive power (Issue #72)



Hello jj,

I currently only have an antenna for these bands

Monday, September 30, 2024, 10:40:30 AM, you wrote:





How many watts are you seeing at these levels?
Drive=50 at 7MHz   = 15W
Drive=85 at 14MH  = 20W
Drive=95 at 21MHz = 10W
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>









-- 
Best regards,
Molnar Gyula

**Comment by drexjj** [comment]
We might be able to read and set the last used drive level per each band that will accomplish this.

**Comment by MolKa51** [comment]
Re: [drexjj/sbitx] QRO drive power (Issue #72)



Hello jj,

I hope this works for someone else. I look forward to the developments.

Tuesday, October 1, 2024, 5:54:58 PM, you wrote:





We might be able to read and set the last used drive level per each band that will accomplish this.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
Best regards,
Molnar Gyula

**Comment by drexjj** [comment]
Added to PR #91 for the next sub-release.

**Comment by MolKa51** [comment]
Re: [drexjj/sbitx] QRO drive power (Issue #72)



Hello jj,

I installed it and tried it and restarted it but I don't see any change.

Monday, October 7, 2024, 3:08:00 AM, you wrote:





Added to PR #91 for the next sub-release.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
Best regards,
Molnar Gyula

**Comment by MolKa51** [comment]
By adjusting the scale value, I adjust the output power, but I cannot return to the original. In the state before QRO, the original scale value must remain, while in the QRO state, a different scale value can be used, but how can it be restored if I don't want to use QRO? I think you should first record the original well-tuned state per band, which is the current scale value. If possible, the value changed in use should be able to be saved. Then record a QRO scale value per band and this would be the scale value recording the output power in the case of QRO.
The conversion should be solved with the QRO switch.
qm-bs currently modifies the scale value, which is an existing thing.
The change for me is if I can change this at QRO and it can be called up automatically.
If it is not QRO, then keep the original scale value, which is the maximum that can be obtained from the hardware or the value reduced by the user.
In summary: the current scale value is the maximum allowed by the hardware at the time of initial setup. If you want to reduce this during use, you must save the scale value for each band so that it works with it the next time you turn it on. The next one is the even further reduced scale value, which can be called up with the QRO switch. The QRO and the usage scale should be switchable back and forth.

**Comment by MolKa51** [comment]
qm-if works well and meets the usage setting.
Record Drive value per band. It's fine even after a reboot. Great !
From here you can proceed to QRO.

**Comment by MolKa51** [comment]
Re: [drexjj/sbitx] QRO drive power (Issue #72)



Hello jj,

Now testing qm-if and this work well.
qm-if works well and meets the usage setting.
Record Drive value per band. It's fine even after a reboot. Great !
From here you can proceed to QRO. 

Monday, October 7, 2024, 3:08:00 AM, you wrote:





Added to PR #91 for the next sub-release.
—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>




-- 
Best regards,
Molnar Gyula

**Comment by drexjj** [comment]
Added in v4.1
